Automattic/wp-calypso
# i18n: Migrate from @automattic/react-i18n to @wordpress/react-i18n

#### Changes proposed in this Pull Request

* Use `@wordpress/react-i18n` in `CalypsoI18nProvider` and Gutenboarding's `LocaleContext` components.
* Replace `@automattic/react-i18n` imports with `@wordpress/react-i18n` in all dependant components.
* Update dependencies.
* Resolve type conflicts in `@automattic/react-i18n` package due to type def changes in `@wordpress/i18n@^3.19.0`.

#### Testing instructions

* Review code changes - `CalypsoI18nProvider`, packages dependencies updates, etc.
* Test Gutenboarding and confirm translations are loaded and displayed as expected.
* Test ETK (Editor Site Launch) and confirm translations are loaded and displayed as expected.
* Test translations in packages:
  * Composite Checkout - can be tested on `calypso.live/checkout/{site}`
  * Onboarding - can be tested on `calypso.live/checkout/{site}`, the action buttons `Skip for now`, `Go back`
  * Language Picker & Search - can be tested on `calypso.live/me/account` (Interface language) or `calypso.live/new/language-modal`
  * Domain picker - can be tested on `calypso.live/new/domains`
  * Plans Grid - can be tested on `calypso.live/new/plans`
  * What's New - can be tested on `calypso.live/post/{site}/new`, Options Menu (three dots button in the top bar) -> What's new
  * Focused Launch - can be tested on `calypso.live/page/{site}/{pageid}`, where `{site}` is a non-launched site
